modules/apps/headless/headless-admin-content/headless-admin-content-impl/src/main/java/com/liferay/headless/admin/content/internal/graphql/extension/StructuredContentQueryContributor.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.admin.content.internal.graphql.extension;
+
+import com.liferay.headless.admin.content.dto.v1_0.StructuredContentVersionInformation;
+import com.liferay.headless.admin.content.dto.v1_0.VersionInformation;
+import com.liferay.headless.delivery.dto.v1_0.StructuredContent;
+import com.liferay.portal.vulcan.graphql.annotation.GraphQLField;
+import com.liferay.portal.vulcan.graphql.annotation.GraphQLTypeExtension;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Luis Miguel Barcos
+ */
+@Component(immediate = true, service = {})
+public class StructuredContentQueryContributor {
+
+	@GraphQLTypeExtension(StructuredContent.class)
+	public class StructuredContentVersionInformationTypeExtension {
+
+		public StructuredContentVersionInformationTypeExtension(
+			StructuredContent structuredContent) {
+
+			_structuredContent = structuredContent;
+		}
+
+		@GraphQLField
+		public VersionInformation versionInformation() {
+			StructuredContentVersionInformation
+				structuredContentVersionInformation =
+					(StructuredContentVersionInformation)_structuredContent;
```

Comment:
We talk about removing the cast to avoid breaking when the call is made from Headless delivery API, could be?

Reply:
Be careful about this one especially

Reply:
Yes, you're right, I forgot it in the refactor. Thanks :)